cluster/addons/iso-client/coreaffinity/isolator.go

```diff
@@ -81,80 +80,99 @@ func getPod(bytePod []byte) (pod *v1.Pod, err error) {
 	return
 }
 
-// check wheter pod should be isolated
-func isIsoPod(pod *v1.Pod) bool {
-	if pod.Annotations["pod.alpha.kubernetes.io/isolation-api"] != "" {
-		return true
+func (i *isolator) gatherContainerRequest(container v1.Container) int64 {
+	resource, ok := container.Resources.Requests[v1.OpaqueIntResourceName(i.Name)]
+	if !ok {
+		return 0
 	}
-	return false
+	return resource.Value()
 }
 
-// extract isoSpec from annotations
-func getIsoSpec(annotations map[string]string) (spec *isoSpec, err error) {
-	spec = &isoSpec{}
-	err = json.Unmarshal([]byte(annotations["pod.alpha.kubernetes.io/isolation-api"]), spec)
-	if err != nil {
-		glog.Fatalf("Cannot unmarshal isoSpec: %v", err)
-		return nil, err
+func (i *isolator) countCoresFromOIR(pod *v1.Pod, cgroupInfo *lifecycle.CgroupInfo) int64 {
+	var coresAccu int64
+	for _, container := range pod.Spec.Containers {
+		coresAccu = coresAccu + i.gatherContainerRequest(container)
 	}
-	return
+	return coresAccu
 }
 
-// validate isoSpec
-func validateIsoSpec(spec *isoSpec) (err error) {
-	if spec.CoreAffinity == "" {
-		return fmt.Errorf("Required field core-affinity is missing.")
+func (i *isolator) eventReplyGenerator(error string, cgroupInfo *lifecycle.CgroupInfo, cgropupResources []*lifecycle.CgroupResource) *lifecycle.EventReply {
+	return &lifecycle.EventReply{
+		Error:          error,
+		CgroupInfo:     cgroupInfo,
+		CgroupResource: cgropupResources,
 	}
-	return nil
+}
+
+func (i *isolator) reserveCores(cores int64) ([]int, error) {
```

Comment:
General comment: we should be consistent in terminology across the code w.r.t. CPUs vs cores. Proposal: `cpu` should always mean logical CPUs as reported by the operating system, `core` should always mean a physical core (possibly hosting two hyperthreads).